src/categorical_algebra/CSets.jl

```diff
@@ -224,48 +227,36 @@ acsets, loose morphisms are usually preferable. For more information about
 limits and colimits in these categories, see [`TightACSetTransformation`](@ref)
 and [`LooseACSetTransformation`](@ref).
 """
-abstract type ACSetTransformation{S<:TypeLevelSchema,Comp,
-                                  Dom<:StructACSet{S},Codom<:StructACSet{S}} end
-# FIXME: The components `Comp` shouldn't be a type parameter in this abstract
-# type but for now it is retained for backwards compatibility.
-
-ACSetTransformation(components, X::StructACSet{S}, Y::StructACSet{S}) where S =
-  ACSetTransformation{S}(components, X, Y)
-ACSetTransformation(X::StructACSet{S}, Y::StructACSet{S}; components...) where S =
-  ACSetTransformation{S}((; components...), X, Y)
-
-function ACSetTransformation{S}(components, X::StructACSet{S}, Y::StructACSet{S}) where S
-  ob_components = filter(∈(objects(S))∘first, pairs(components))
-  type_components = filter(∈(attrtypes(S))∘first, pairs(components))
-  length(ob_components) + length(type_components) == length(components) ||
-    error("Not all names in $(keys(components)) are objects or attribute types")
-  if isempty(type_components)
-    TightACSetTransformation{S}(ob_components, X, Y)
-  else
-    LooseACSetTransformation{S}(ob_components, type_components, X, Y)
-  end
-end
+abstract type ACSetTransformation end
+abstract type TightACSetTransformation <:ACSetTransformation end
+abstract type LooseACSetTransformation <:ACSetTransformation end
+
 
 components(α::ACSetTransformation) = α.components
 force(α::ACSetTransformation) = map_components(force, α)
 
-""" Transformation between C-sets.
 
-Recall that a C-set homomorphism is a natural transformation: a transformation
-between functors C → Set satisfying the naturality axiom for every morphism, or
-equivalently every generating morphism, in C.
+# Dynamic ACSet transformations
+@struct_hash_equal struct DynamicTightACSetTransformation <: TightACSetTransformation
+  components::NamedTuple
```

Comment:
Not the most important thing, but IMO the most natural type for `components` would be `AbstractDict{Symbol}` rather than `NamedTuple` because in the latter, the key names are static (contained in a type parameter of `NamedTuple`).

Reply:
There are some issues with code that is current generic w/r/t `Struct` vs `Loose` that gets broken if this is changed to an `AbstractDict`, so perhaps this can be changed in a later PR?

Reply:
Postponed to #750